UserAgent/Views/Privacy Indicator/PrivacyDashboardView.swift

```diff
@@ -1,245 +1,124 @@
-
-        titleStackView.snp.makeConstraints { make in
+        view.snp.makeConstraints { make in
             make.top.leading.trailing.equalToSuperview()
         }
-
-        mainStackView.snp.makeConstraints { make in
-            make.top.equalTo(titleStackView.snp.bottom).offset(16)
-            make.leading.trailing.bottom.equalToSuperview()
-        }
-
-        privacyIndicator.snp.makeConstraints { make in
-            make.width.equalTo(mainStackView).dividedBy(2.2)
-        }
-
-        numberOfTrackersLabel.snp.makeConstraints { make in
-            make.center.equalTo(privacyIndicator)
-        }
+        return view
     }
 
-    func circleView(withColor: UIColor) -> UIView {
-        let radius = 5
-        let topMargin = 4
+    func renderMain(_ wrapper: UIView, _ header: UIView) {
+        let subviews = [self.privacyIndicator, wrapper]
+        let view = UIStackView(arrangedSubviews: subviews)
+        view.spacing = 10
+        view.alignment = .top
+        view.distribution = .fillProportionally
+        self.addSubview(view)
 
-        let containerView = UIView()
-        containerView.snp.makeConstraints { make in
-            make.width.equalTo(radius * 2)
-            make.height.equalTo((radius * 2) + (topMargin * 2))
+        view.snp.makeConstraints { make in
+            make.top.equalTo(header.snp.bottom).offset(16)
+            make.leading.trailing.bottom.equalToSuperview()
         }
-
-        let circleView = UIView()
-        containerView.addSubview(circleView)
-        circleView.snp.makeConstraints { make in
-            make.width.equalTo(circleView.snp.height)
-            make.height.equalTo(radius * 2)
-            make.center.equalTo(containerView)
+        self.privacyIndicator.snp.makeConstraints { make in
+            make.width.equalTo(view).dividedBy(2.2)
         }
+    }
 
-        circleView.backgroundColor = withColor
-        circleView.layer.cornerRadius = CGFloat(radius)
+    func renderWrapper() -> UIStackView {
+        let view = UIStackView()
+        view.alignment = .top
+        view.axis = .vertical
+        view.spacing = 5
+        return view
+    }
 
-        return containerView
+    func renderStats(_ wrapper: UIStackView, _ stats: TPPageStats) {
+        let statsDict = WTMCategory.statsDict(from: stats)
+        WTMCategory.all()
+            .filter { statsDict[$0, default: 0] != 0 }
+            .forEach { self.renderStat(wrapper, $0, statsDict) }
     }
 
-    func statLabel(labelled text: String) -> UILabel {
-        let label = UILabel()
-        label.textColor = Theme.textField.textAndTint
-        label.text = text
-        label.numberOfLines = 0
-        label.setContentCompressionResistancePriority(.required, for: .horizontal)
-        label.font = UIFont.preferredFont(forTextStyle: .footnote)
-        return label
+    func renderStat(
+        _ wrapper: UIStackView,
+        _ category: WTMCategory,
+        _ statsDict: [WTMCategory: Int]
```

Comment:
I do like a lot this multi-line formatting.
I'm kinda wondering why you avoid the usage of named arguments (or however this is called in Swift :P)? In a way named arguments make function calls more expressive (Smalltalk style). AFAIK it comes from objective-c, right?

Reply:
have no idea :) 
Yeap, you are right, make sense to name it :)